### PR TITLE
Add hal-flash package

### PR DESCRIPTION
**Pull request by malyn**
I have a change to enable this in firefoxWrapper and will send a PR for that after this one is on master.

### CODE REVIEW

File: pkgs/os-specific/linux/hal-flash/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  meta = with stdenv.lib; {
+    homepage = https://github.com/cshorler/hal-flash;
+    description = "libhal stub library to satisfy the Flash Player DRM requirements.";
+    longDescription =
+      ''
+        Stub library based loosely upon libhal.[ch] from the hal-0.5.14
+        package.  Provides the minimum necessary functionality to enable
+        libflashplayer.so/libadobecp.so to play back DRM content.
+      '';
+    license = with licenses; [ afl21 gpl2 ];
```
Comment (peti) [inline pkgs/os-specific/linux/hal-flash/default.nix]: `meta.license` is not supposed to be a list: https://nixos.org/nixpkgs/manual/#idm140737318982464.

Comment (malyn) [inline pkgs/os-specific/linux/hal-flash/default.nix]: Yeah, I was wondering about that...  What's the right way to handle this situation?  Drop the license property, pick a license, make it a string that contains the license names, ..?  I've seen all of these (and the list, sadly!) in other nix expressions in nixpkgs.

**Comment by malyn** [comment]
Any thoughts on how to handle the dual-license situation?  I have a similar issue in PR #7494, which is a package that contains multiple fonts with multiple licenses.

File: pkgs/os-specific/linux/hal-flash/default.nix (hunk 1 above)
Comment (malyn) [inline pkgs/os-specific/linux/hal-flash/default.nix]: Per a discussion in #7494, I will replace this list with a string containing the spdxIds for the licenses.

Comment (malyn) [inline pkgs/os-specific/linux/hal-flash/default.nix]: Slight change, will use the shortname instead.

**Comment by malyn** [comment]
I fixed the license property; it's now a space-separated string of license names.

**Pull request merged**